### PR TITLE
Fix prog/main_test.go

### PR DESCRIPTION
**Pull request by rndstr**
Is excluded from tests and does not compile.

### CODE REVIEW

**Comment by rndstr** [comment]
Not sure what the consensus is on testing private interfaces; if "okay" then I can re-add and fix it

**Comment by 2opremio** [comment]
Good catch, sigh. I think we should re-add it.

**Comment by rndstr** [comment]
@2opremio ok if I use github.com/stretchr/testify/assert? It's already in `vendor/`

**Comment by 2opremio** [comment]
Sure

**Comment by rndstr** [comment]
Executing any backend tests in `prog/` requires the "scope ui asset providers" (`staticui` & `externalui`) to be generated. These in turn have a [dependency on the client app](https://github.com/weaveworks/scope/blob/7d8448c586ea2743815558fab70b2f593bbc0fd5/Makefile#L84-L85) which then needs to be fully built. For this to work we would need to move the `scope-ui-backend` docker image to a node based image (or additionally install node && npm). Which is in my opinion not worth the trouble till we actually do have asset-delivery tests.

File: Makefile [hunk 1]
```diff
@@ -179,10 +179,12 @@ tmp/weave-scope.tgz: $(shell find client/app -type f) $(SCOPE_UI_BUILD_UPTODATE)
 else
 
 client/build/index.html:
-	cd client && npm run build
+	test "true" = "$(SCOPE_SKIP_UI_ASSETS)" && { mkdir -p client/build; } || \
```
Comment (rndstr) [inline Makefile]: I'm pretty sure the first expression could be improved?

**Comment by 2opremio** [comment]
> Which is in my opinion not worth the trouble till we actually do have asset-delivery tests.

I agree. I didn't know this was going to open a can of worms.

File: Makefile (hunk 1 above)
Comment (2opremio) [inline Makefile]: I've never seen braces used like this. Is it for subprocess execution? What's the difference between `{ mkdir -p client/build; }` and `( mkdir -p client/build; )`?

Comment (rndstr) [inline Makefile]: You are right, there is no difference in what is executed, I removed them. In the subsequent line I think keeping it makes sense for improved readability.

**Comment by rndstr** [comment]
@2opremio I was wrong with `Edit2`, should work as intended & ready for review :)

File: tools/test [hunk 2]
```diff
@@ -74,7 +74,7 @@ fail=0
 
 if [ -z "$TESTDIRS" ]; then
     # NB: Relies on paths being prefixed with './'.
-    TESTDIRS=($(git ls-files -- '*_test.go' | grep -vE '^(vendor|prog|experimental)/' | xargs -n1 dirname | sort -u | sed -e 's|^|./|'))
+    TESTDIRS=($(git ls-files -- '*_test.go' | grep -vE '^(vendor|experimental)/' | xargs -n1 dirname | sort -u | sed -e 's|^|./|'))
```
Comment (bboreham) [inline tools/test]: This change needs to be made in the build-tools repository

Comment (rndstr) [inline tools/test]: I stumbled over this: https://github.com/weaveworks/build-tools/pull/103/files#diff-098f6bcd4621d373cade4e832627b4f6R81

Is this sufficient?

Comment (bboreham) [inline tools/test]: Yes, by mistake that PR included your change.

**Pull request merged**